kernel/src/boot_stage2.rs

```diff
@@ -98,19 +110,18 @@ global_asm!(
         orl %edx, %eax
         movl %eax, 0xF6C(%edi)
 
-        /* Signal APs */
-        movl $setup_flag, %edi
-        movl $1, (%edi)
         jmp 2f
 
-.Lskip_paging_setup:
-        movl $setup_flag, %edi
-.Lap_wait:
+    .Lskip_paging_setup:
+        movl $ap_flag, %edi
+    .Lap_wait_for_env:
         movl (%edi), %eax
         test %eax, %eax
-        jz .Lap_wait
+        jnz 2f
+        pause
+        jmp .Lap_wait_for_env
```

Comment:
```suggestion
        pause
        jz .Lap_wait_for_env
```

There's nothing wrong with executing `PAUSE` if the loop isn't taken; `PAUSE` hint exiting occurs when the same `PAUSE` is run too many times in a row.  If you follow my suggestion, you can drop an extra branch and make the code tighter.

Reply:
Makes sense. I'll do that.